pkg/descheduler/strategies/duplicates.go

```diff
@@ -46,12 +55,43 @@ func RemoveDuplicatePods(ds *options.DeschedulerServer, strategy api.Descheduler
 // deleteDuplicatePods evicts the pod from node and returns the count of evicted pods.
 func deleteDuplicatePods(client clientset.Interface, policyGroupVersion string, nodes []*v1.Node, dryRun bool, nodepodCount utils.NodePodEvictedCount, maxPodsToEvict int, evictLocalStoragePods bool) int {
 	podsEvicted := 0
+	cepm := creatorEvictedPods{}
 	for _, node := range nodes {
 		klog.V(1).Infof("Processing node: %#v", node.Name)
 		dpm := ListDuplicatePodsOnANode(client, node, evictLocalStoragePods)
 		for creator, pods := range dpm {
+			if _, found := cepm[creator]; !found {
+				cepm[creator] = 0
+			}
+			status, err := getStatusForOwnerRef(client, creator)
+			if err != nil {
+				klog.Errorf("Error getting the number of replicas: %v", err)
+				continue
+			}
+			// bestRatio := float64(status.replicas) / float64(len(nodes))
+			bestRatio := (int(status.replicas) / len(nodes)) + 1
+			// here we have to implement the logic to skip eviction
+			// 1. too few available replicas
+			klog.V(3).Infof("%#v with only %.2f%% of available pods, skipping?", creator, (float64(status.available)/float64(status.replicas))*100.0)
+			if (float64(status.available) / float64(status.replicas)) < 0.6 {
+				klog.V(1).Infof("%#v with only %.2f%% of available pods, skipping", creator, (float64(status.available)/float64(status.replicas))*100.0)
+				continue
+			}
+			// 2. in case the replicas is greater than the number of nodes and we are enough close to the best balance
+			klog.V(3).Infof("%#v on this node has close enough to the best ratio (%d) number of pods %d over replicas %d, skipping?", creator, bestRatio, len(pods), status.replicas)
+			if len(pods) <= (bestRatio + 1) {
+				klog.V(1).Infof("%#v on this node has close enough to the best ratio (%d) number of pods %d over replicas %d, skipping", creator, bestRatio, len(pods), status.replicas)
+				continue
+			}
+			// 3. we already evicted some pods for this creator from other nodes
+			klog.V(3).Infof("%#v with %d available replicas, we already evicted %d pods, skipping?", creator, status.available, cepm[creator])
+			if (float64(cepm[creator]) / float64(status.available)) >= 0.1 {
```

Comment:
Conditions that holds at this point:
- `cepm[creator]` <= `status.available` (you can't evict more than you have available)
- `status.available` > 0 (you need at least one instance otherwise `dpm[creator]` is empty)

 Your inequality says we skip when the number of evicted pods for `creator` is at least 1/10 of available. Why `0.1`? Why not `bestRatio`? I.e. `(status.available - cepm[creator]) < bestRatio`?

Also, you might move this condition under `for i := 1; i < len(pods); i++ {` instead. So you don't evict more than is guarded by `bestRatio`.